src/goLint.ts

```diff
@@ -40,9 +40,10 @@ export function lintCode(lintWorkspace?: boolean) {
  *
  * @param fileUri Document uri.
  * @param goConfig Configuration for the Go extension.
- * @param lintWorkspace If true runs linter in all workspace.
+ * @param lintWorkspace If true, runs linter in the entire workspace.
+ * @param lintFile If true, runs linter on a single file.
  */
-export function goLint(fileUri: vscode.Uri, goConfig: vscode.WorkspaceConfiguration, lintWorkspace?: boolean): Promise<ICheckResult[]> {
+export function goLint(fileUri: vscode.Uri, goConfig: vscode.WorkspaceConfiguration, lintWorkspace?:boolean, lintFile?: boolean): Promise<ICheckResult[]> {
```

Comment:
Using 2 separate variables to manage scope feels excessive. Why not use a single variable, say `scope`? This can have one of the 4 values `file`, `package`, `workspace`, `off`.

Reply:
Thanks! I was wondering about this. I took your advice and passed a string value around. It cleans up the signatures. 

I appreciate the feedback 😄